focus-ios/Blockzilla/Modules/WebView/LegacyWebViewController.swift

```diff
@@ -535,15 +535,15 @@ extension LegacyWebViewController: WKUIDelegate {
         // check if this is a new frame / window
         guard navigationAction.targetFrame == nil else { return nil }
 
-        // validate the URL using URIFixup
-        guard let urlString = navigationAction.request.url?.absoluteString,
-              URIFixup.getURL(entry: urlString) != nil else {
-            // URL failed validation, prevent loading
+        // If URL is a file:// when web application calls window.open() or fails validation, prevent loading
+        guard let url = navigationAction.request.url,
+              let validatedURL = URIFixup.getURL(entry: url.absoluteString),
+              url.scheme != "file" else {
```

Comment:
Do we know if there are any scenarios in which the `file://` scheme should be allowed? I'm not familiar with all of the potential use cases for that.

Reply:
Hi, wouldn't this stop from opening file type? 

This solution basically stops the file from opening a new tab but that isn't something we should prevent. Instead we should be looking to make sure that the url bar has the correct address from the file `open` method.

Reply:
hey @nbhasin2 I'm also not familiar how file:// urls should behave on Focus. Do we have a use case for Firefox? I guess my question is how we can differentiate between a malicious file url and a good one.